client/sources/scheme_test/models.py

```diff
@@ -0,0 +1,177 @@
+"""Interprets Scheme test files and compares each line of printed output from
+the read-eval-print loop and from any output functions to an expected output
+described in a comment.  For example,
+
+(display (+ 2 3))
+; expect 5
+
+Differences between printed and expected outputs are printed with line numbers.
+
+Requires the following attributes from an importable scheme module:
+
+    scheme.create_global_frame
+    scheme.Buffer
+    scheme.read_eval_print_loop
+    scheme.tokenize_lines
+"""
+
+from client.sources.common import models
+from client.utils import format
+from client.utils import output
+from client.utils import timer
+import importlib
+import re
+import sys
+
+
+##########
+# Models #
+##########
+
+class SchemeTest(models.Test):
+
+    def __init__(self, file, file_contents, timeout=None, **fields):
+        super().__init__(**fields)
+        self.file = file
+        self.file_contents = file_contents
+        self.timeout = timeout
+
+    def run(self):
+        """Runs the suites associated with this doctest.
+
+        RETURNS:
+        bool; True if the doctest completely passes, False otherwise.
+        """
+        format.print_line('-')
+        print('Scheme tests in {}'.format(self.file))
+        print()
+
+        passed, failed = self._run_tests()
+
+        print('{} passed; {} failed'.format(passed, failed))
+        if failed == 0 and passed > 0:
+            print('-- OK! --')
+            print()
+
+        return {'passed': passed, 'failed': failed, 'locked': 0}
+
+    def score(self):
+        format.print_line('-')
+        print('Scheme tests in {}'.format(self.file))
+        print()
+        _, failed = self._run_tests()
+        score = 1.0 if failed == 0 else 0.0
+
+        print('Score: {}/1'.format(score))
+        print()
+        return score
+
+    def unlock(self, interact):
```

Comment:
I recall originally having them in the `Test` object, but you recommended the `Test` object methods just raise `NotImplementedError`s.